### ISSUE TITLE
InterfaceType in query added twice

### ISSUE THREAD

**Issue by batbeer** [ChilliCream/graphql-platform#6589]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Product

Hot Chocolate

### Describe the bug

it tries to add the InterfaceType both from the query and the type itself, causing hotchocolate to get an error: 

The name `Test` was already registered by another type. (HotChocolate.Types.ObjectType<Path.To.Class.Test>)

### Steps to reproduce

1. Define abstract class

public abstract class Test
{
    public string Name { get; set; }
}

public class TestText : Test
{
    public string Text { get; set; }
}

public class TestNumber : Test
{
    public decimal Number { get; set; }
}
public class TestType : InterfaceType<Test>
{
    protected override void Configure(IInterfaceTypeDescriptor<Test> descriptor)
    {
        descriptor.BindFieldsExplicitly();

        descriptor.Field(f => f.Name);
    }
}

public class TestTextType : ObjectType<TestText>
{
    protected override void Configure(IObjectTypeDescriptor<TestText> descriptor)
    {
        descriptor.BindFieldsExplicitly();

        descriptor.Name("TestText");

        descriptor.Implements<TestType>();

        descriptor
            .Field(f => f.Name);

        descriptor
            .Field(f => f.Text);
    }
}

public class TestNumberType : ObjectType<TestNumber>
{
    protected override void Configure(IObjectTypeDescriptor<TestNumber> descriptor)
    {
        descriptor.BindFieldsExplicitly();

        descriptor.Name("TestNumber");

        descriptor.Implements<TestType>();

        descriptor
            .Field(f => f.Name);

        descriptor
            .Field(f => f.Number);
    }
}

2. Use in query

[ExtendObjectType(OperationTypeNames.Query)]
public class TestQuery
{
    [UseDbContext(typeof(DatabaseContext))]
    [GraphQLDescription("Get Test")]
    [UsePaging(IncludeTotalCount = true, DefaultPageSize = 20, MaxPageSize = 50)]
    [UseProjection]
    [UseSorting(typeof(TestSortInput))]
    [UseFiltering(typeof(TestFilterInput))]
    public IQueryable<Test> GetTests(DatabaseContext databaseContext)
    {
        //here the abstract class is added as an objectType, then causing the error
        //return databaseContext.Tests
    }
}

public class TestFilterInput : FilterInputType<Test>
{
    protected override void Configure(IFilterInputTypeDescriptor<Test> descriptor)
    {
        descriptor.BindFieldsExplicitly();

        descriptor.Field(f => f.Name);
    }
}

public class TestSortInput : SortInputType<Test>
{
    protected override void Configure(ISortInputTypeDescriptor<Test> descriptor)
    {
        descriptor.BindFieldsExplicitly();

        descriptor.Field(f => f.Name);
    }
}

3. run program

result in the error: 
The name `Test` was already registered by another type. (HotChocolate.Types.ObjectType<Path.To.Class.Test>)

### Relevant log output

```shell
The name `Test` was already registered by another type. (HotChocolate.Types.ObjectType<Path.To.Class.Test>)
```

### Version

13.5.1

**Comment by PascalSenn** [comment]
Do you register the InterfacetTYpe on the schema builder with `AddType<TestType>` ?

**Comment by batbeer** [comment]
yes it is registered, the issue is that it tries to register it again as a objectType when used in the query, dont know why it hides this in the example too: public IQueryable<Test> GetTests(DatabaseContext databaseContext)

**Comment by PascalSenn** [comment]
Can you provide a minimal reproduction of this issue?

**Comment by madiganz** [comment]
@batbeer trying add `UsePaging(typeof(TestType))`. I had the same problem as you and just figured it out. I believe it's do to something in the `UsePaging` logic not able to find the interface type correctly.

**Comment by batbeer** [comment]
@madiganz Thanks! this works 👍

**Comment by michaelstaib** [comment]
Rechecked on main and could not reproduce this anymore.

I validated a minimal setup with:
- an abstract runtime type used as the query return (`IQueryable<Test>`)
- explicit interface/object registrations (`AddType<TestType>`, `AddType<TestTextType>`, `AddType<TestNumberType>`)
- paging + projection + sorting + filtering middleware

Schema creation succeeds without duplicate type registration errors.

Closing this as non-reproducible on current main. If this still reproduces on the latest version, please reopen with an updated minimal repro and exact package versions.
